server/app/views/applicant/NorthStarApplicantCommonIntakeUpsellView.java

```diff
@@ -0,0 +1,140 @@
+package views.applicant;
+
+import auth.CiviFormProfile;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import controllers.AssetsFinder;
+import controllers.LanguageUtils;
+import controllers.applicant.ApplicantRoutes;
+import java.util.ArrayList;
+import java.util.Locale;
+import modules.ThymeleafModule;
+import org.thymeleaf.TemplateEngine;
+import play.i18n.Messages;
+import play.mvc.Http.Request;
+import services.DeploymentType;
+import services.applicant.ApplicantPersonalInfo;
+import services.applicant.ApplicantService.ApplicantProgramData;
+import services.settings.SettingsManifest;
+
+public class NorthStarApplicantCommonIntakeUpsellView extends NorthStarApplicantBaseView {
+
+  @Inject
+  NorthStarApplicantCommonIntakeUpsellView(
+      TemplateEngine templateEngine,
+      ThymeleafModule.PlayThymeleafContextFactory playThymeleafContextFactory,
+      AssetsFinder assetsFinder,
+      ApplicantRoutes applicantRoutes,
+      SettingsManifest settingsManifest,
+      LanguageUtils languageUtils,
+      DeploymentType deploymentType) {
+    super(
+        templateEngine,
+        playThymeleafContextFactory,
+        assetsFinder,
+        applicantRoutes,
+        settingsManifest,
+        languageUtils,
+        deploymentType);
+  }
+
+  public String render(Params params) {
+    ThymeleafModule.PlayThymeleafContext context =
+        createThymeleafContext(
+            params.request(),
+            params.applicantId(),
+            params.profile(),
+            params.applicantPersonalInfo(),
+            params.messages());
+
+    context.setVariable("applicationId", params.applicationId());
+    context.setVariable("messages", params.messages());
+
+    String linkHref = settingsManifest.getCommonIntakeMoreResourcesLinkHref(params.request()).get();
+    String linkText = settingsManifest.getCommonIntakeMoreResourcesLinkText(params.request()).get();
+    String linkHtml =
```

Comment:
you should be able to just pass in the linkRef and linkText to thymeleaf and generate the HTML in thymeleaf. It can be a fragment that you call like here https://github.com/civiform/civiform/blob/649320018030ed04b6b663910eed2b342fe38e90/server/app/views/applicant/ApplicantBaseFragment.html#L16

Reply:
This line works:
```
<div th:replace="~{applicant/ApplicantBaseFragment :: hyperlink(${linkHref}, ${linkText})}"></div>
```

This line results in the link text being 'null'
```
<p th:text="#{content.commonIntakeNoMatchingPrograms( ~{applicant/ApplicantBaseFragment.hyperlink(${linkHref},  ${linkText})} )}"></p>
```

According to my research on StackOverflow and Gemini, there's no great way to solve this in Thymeleaf.

The ideal solution is to change the UX spec to not have links embedded in translated strings. e.g. Have a button that shows ${linkText} separate from the <p>

Do you have any other ideas?

Reply:
nope that seems fine. can you just add a comment?